### PR TITLE
Python. Homework #2. Task #1. Власов Илья

### PR DESCRIPTION
**Pull request by GirZ0n**

### CODE REVIEW

**Comment by nbirillo** [comment]
Should I review this PR since you have a TODO?

File: homework/homework2/task1/curry.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from typing import Callable, Any
+
+
+def curry_explicit(function: Callable, arity: int) -> Callable:
```
Comment (nbirillo) [inline homework/homework2/task1/curry.py:4]: Please, try to fix your functions for this case:
```
f4 = curry_explicit(print, 4)
g4 = uncurry_explicit(f4, 4)
f2 = f4(10)(20)
g2 = uncurry_explicit(f2, 2)
print('---')
f4(5)(15)(25)(35)
f2(30)(40)
print('---')
g4(5, 15, 25, 35)
g2(30, 40)
```
The expected output in this case:
```
---
5 15 25 35
10 20 30 40
---
5 15 25 35
10 20 30 40
```
But in your case, you will have an error

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great! I have only one comment

File: homework/homework2/task1/curry.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from typing import Callable, Any, List
+
+
+def curry_explicit(function: Callable, arity: int) -> Callable:
```
Comment (nbirillo) [inline homework/homework2/task1/curry.py:4]: Sorry, but I found another bug:
```
f0 = curry_explicit(lambda *xs: print(xs), 0)

print('try nullary:')
f0()

try:
    print('try unary:')
    f0(1)
except Exception as e:
    print('-- expected exception.')

try:
    print('try nullary-nullary:')
    f0()()
except Exception as e:
    print('-- expected exception.')
```
The expected output is:
```
try nullary:
()
try unary:
-- expected exception.
try nullary-nullary:
()
-- expected exception.
```
Your output is:
```
try nullary:
()
try unary:
(1,)
try nullary-nullary:
()
-- expected exception.
```

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great! I hope it is the last comment

**Review by SpirinEgor — APPROVED** [review]

**Pull request merged**